src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -185,9 +188,10 @@
         Ref("NullLiteralSegment"),
         Ref("DateTimeLiteralGrammar"),
         Ref("ParameterNameSegment"),
+        Ref("SystemVariableSegment"),
     ),
     ParameterNameSegment=RegexParser(
-        r"[@][A-Za-z0-9_]+", CodeSegment, name="parameter", type="parameter"
+        r"@[A-Za-z0-9_]+", CodeSegment, name="parameter", type="parameter"
```

Comment:
Not related to this, but noticed it while where and no need for square brackets in first section.